folium/plugins/heat_map_withtime.py

```diff
@@ -0,0 +1,245 @@
+from branca.utilities import none_min, none_max
+
+from folium.map import TileLayer
+
+class HeatMapWithTime(TileLayer):
+    def __init__(self, data, index=None, times=None, name=None, radius=15, max_opacity=0.6, scale_radius=False,
+                 use_local_extrema=False, default_weight=1, overlay=True,
+                 auto_play=False, backward_button=True, display_date=True,
+                 forward_button=True, limit_minimum_range=5, limit_sliders=False,
+                 loop_button=False, min_speed=0.1, max_speed=10, play_button=True,
+                 play_reverse_button=False, position="bottomleft", speed_slider=True,
+                 speed_step=0.1, style_NS = "leaflet-control-timecontrol", time_slider=True,
+                 time_slider_drap_update=False, time_steps=1, time_control_title="Time Control"
+                 ):
+        """Create a HeatMapWithTime layer
+
+        Parameters
+        ----------
+
+        """
+        super(TileLayer, self).__init__(name=name)
+        self._name = 'HeatMap'
+        self._control_name = self.get_name() + 'Control'
+        self.tile_name = name if name is not None else self.get_name()
+
+        # input data
+        self.data = data
+        self.index = index if index is not None else [str(i) for i in range(1, len(data)+1)]
+        self.times = times if times is not None else range(1, len(data)+1)
+
+        # heatmap settings
+        self.radius = radius
+        self.max_opacity = max_opacity
+        self.scale_radius = "true" if scale_radius else "false"
+        self.use_local_extrema = "true" if use_local_extrema else "false"
+        self.default_weight = default_weight
+        self.overlay = overlay
+
+        # time dimension settings
+        self.auto_play = "true" if auto_play else "false"
+        self.backward_button = "true" if backward_button else "false"
+        self.display_date = "true" if display_date else "false"
+        self.forward_button = "true" if forward_button else "false"
+        self.limit_minimum_range = "true" if limit_minimum_range else "false"
+        self.limit_sliders = "true" if limit_sliders else "false"
+        self.loop_button = "true" if loop_button else "false"
+        self.min_speed = min_speed
+        self.max_speed = max_speed
+        self.play_button = "true" if play_button else "false"
+        self.play_reverse_button = "true" if play_reverse_button else "false"
+        self.position = position
+        self.speed_slider = "true" if speed_slider else "false"
+        self.speed_step = speed_step
+        self.style_NS = style_NS
+        self.time_slider = "true" if time_slider else "false"
+        self.time_slider_drap_update = "true" if time_slider_drap_update else "false"
+        self.time_steps = time_steps
+        self.time_control_title = time_control_title
+
+        self._template = Template(u"""
+        {% macro script(this, kwargs) %}
+
+            var times = {{this.times}};
```

Comment:
In the gist I did, it was :
```javascript
var times = [];
for (var i=0; i<data.length; i++) {times.push(1+i);}
```
The interest was to let the user manage the timeline through the `index` attribute instead of `times`.
Your code is obviously even more agile, but I wonder if it will not seem too complex to the final user.
It may depend on the docstring.

It's just a question ; I let you take the decision.